### PR TITLE
Fix for bad MIME header

### PR DESCRIPTION
**Pull request by haelge**
Set proper content-type, charset, boundary and transfer-encoding.

### CODE REVIEW

File: src/BilloPHPMailer.php [hunk 1]
```diff
@@ -4,20 +4,35 @@
 
 class BilloPHPMailer extends PHPMailer
 {
-  private $_billoContent;
+  private $_billoContent, $_billoHeader;
```
Comment (dschmidt) [inline src/BilloPHPMailer.php]: Nitpick: I'd prefer this on a new line.

File: src/BilloPHPMailer.php [hunk 2]
```diff
@@ -4,20 +4,35 @@
...
-  public function createBody() {
+  
+  public function getMailMIME()
+  {
+    $result = "";
+    foreach($this->_billoHeader as $key => $value){
+       $result.=$this->headerLine($key,$value);
+    }
+    return $result;
+  }
```
Comment (dschmidt) [inline src/BilloPHPMailer.php]: Can you provide a little unit test for this?

File: src/billo.php [hunk 3]
```diff
@@ -32,12 +32,18 @@ function distributeMessage($listsConfig, $mailSender, $id, $message) {
...
       '['.$listsConfig->getListName($list->getEmail()).'] '.$message->subject;
 
-
+    // Extract and set MIME Header
+    $header = [];
+    $header['Content-Type'] = $message->ContentType;
+    if (isset($message->ContentTransferEncoding)){
+        $header['Content-Transfer-Encoding'] = $message->ContentTransferEncoding;
+    }
+    $mailSender->setBilloHeader($header);
```
Comment (dschmidt) [inline src/billo.php]: This is okay, basically what I suggested :-) 

The only nitpick I have now is that we are coming up with a data structure that two parts of the code depend on.  And we are looping twice to generate the array and then to serialize ... maybe we should simply create the string here in place and make the setBilloHeader function even dumber?

With the additional benefit that we could add headers like these multiple times (your array implementation allows only exactly one value per header): 
https://stackoverflow.com/a/18040297/2886832
No idea if we should or need to handle any of these but at least it would give us more flexibility to do so.

Another idea I had was that we could make `getMailMime()` return an empty string and then use `addCustomHeader` but that doesn't allow adding headers with new lines in them, which would require us to fiddle with the header content again which I would like to avoid ... so ... would have been nice but isn't :-D

File: src/BilloPHPMailer.php (hunk 2 above)
Comment (dschmidt) [reply src/BilloPHPMailer.php]: At least if we don't make it dumber as suggested further down...

File: src/billo.php [hunk 4]
```diff
@@ -94,10 +100,9 @@ function main() {
     if ($message->hasFlag(Storage::FLAG_SEEN)) {
       continue;
     } else {
+      distributeMessage($listsConfig, $GLOBALS['mailSender'], $id, $message);
       messageAddFlag($GLOBALS['mailStorage'], $id, $message, Storage::FLAG_SEEN);
     }
-
-    distributeMessage($listsConfig, $GLOBALS['mailSender'], $id, $message);
```
Comment (dschmidt) [inline src/billo.php]: Please undo this change / rebase onto / merge from master.

**Review by dschmidt — CHANGES_REQUESTED** [review]
I already like it a lot, just a few nitpicks and suggestions

File: src/BilloPHPMailer.php (hunk 1 above)
Comment (haelge) [reply src/BilloPHPMailer.php]: I'm  fine with that. I'll change it.

File: src/BilloPHPMailer.php (hunk 2 above)
Comment (haelge) [reply src/BilloPHPMailer.php]: I like the `headerLine()` method, which cares about the correct Fieldname and End of Line. So making it dumber means doing this stuff extra in billo.php (but of course in favor to not loop twice over an array)

File: src/billo.php (hunk 3 above)
Comment (haelge) [reply src/billo.php]: Is it disireable to define a header multiple times? I actually aimed to prevent this XD

Otherwise as suggested in the latest issue, we could just overwrite the `createHeader()` method and simply do what ever we want but this leeds also to more "like to avoid"-stuff.

File: src/BilloPHPMailer.php (hunk 2 above)
Comment (dschmidt) [reply src/BilloPHPMailer.php]: headerLine is public, you can use it here too

File: src/billo.php (hunk 3 above)
Comment (dschmidt) [reply src/billo.php]: Maybe good enough to use getMailMime() for now

**Comment by dschmidt** [comment]
Great! Thanks <3
